cdap-app-fabric/src/main/java/io/cdap/cdap/app/store/Store.java

```diff
@@ -320,6 +320,16 @@ List<ProgramHistory> getRuns(Collection<ProgramReference> programs, ProgramRunSt
    */
   Map<ProgramRunId, RunRecordDetail> getActiveRuns(ProgramId programId);
 
+  /**
+   * Fetches the active (i.e STARTING or RUNNING or SUSPENDED) run records against a given ProgramId.
+   * Ignores the versions of the programs. If multiple versions of the same program are found,
+   * it returns all of them.
+   *
+   * @param programId the program id to match against
+   * @return map of logged runs
+   */
+  Map<ProgramRunId, RunRecordDetail> getActiveRunsAcrossVersions(ProgramId programId);
```

Comment:
Use ProgramReference instead of ProgramId for version-agnostic API

Reply:
+1 to using ProgramReference. Also, if we never use the contents of the Map, it's better to just return the count, as it is slightly less costly in memory and compute.

Reply:
Updated the method to take ProgramReference, and return an int. It does not construct any map anymore, it just increments an atomic int when an active run record is found.